platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java

```diff
@@ -637,34 +638,8 @@ private Collection<SwirldsPlatform> createLocalPlatforms(
                         mainClassName, swirldName, nodeId, appVersion, addressBook.copy(), emergencyRecoveryManager);
 
                 // check software version compatibility
-                final SoftwareVersion loadedSoftwareVersion = loadedSignedState == null
-                        ? null
-                        : loadedSignedState
-                                .getState()
-                                .getPlatformState()
-                                .getPlatformData()
-                                .getCreationSoftwareVersion();
-                final int versionComparison =
-                        loadedSoftwareVersion == null ? 1 : appVersion.compareTo(loadedSoftwareVersion);
-                final boolean softwareUpgrade;
-                if (versionComparison < 0) {
-                    throw new IllegalStateException(
-                            "The current software version `" + appVersion + "` is prior to the software version `"
-                                    + loadedSoftwareVersion + "` that created the state that was loaded from disk.");
-                } else if (versionComparison > 0) {
-                    softwareUpgrade = true;
-                    logger.info(
-                            STARTUP.getMarker(),
-                            "Software upgrade from version {} to {}. ",
-                            loadedSoftwareVersion,
-                            appVersion);
-                } else {
-                    softwareUpgrade = false;
-                    logger.info(
-                            STARTUP.getMarker(),
-                            "No Software Upgrade, continuing with software version {}.",
-                            loadedSoftwareVersion);
-                }
+                final boolean softwareUpgrade =
+                        BootstrapUtils.detectSoftwareUpgrade(appVersion, loadedSignedState, logger);
```

Comment:
Better to have a new logger instance in this static class. Eventually we may end up passing a non-static log manager around, but for now it's best to follow the static log manager pattern.

Reply:
created static logger for the util class and removed from method argument.